packages/rum/src/domain/getSessionReplayLink.ts

```diff
@@ -0,0 +1,55 @@
+import type {
+  RumConfiguration,
+  RumSessionManager,
+  ViewContext,
+  ViewContexts,
+  RumSession,
+} from '@datadog/browser-rum-core'
+import { getDatadogOrigin } from '@datadog/browser-rum-core'
+import { isBrowserSupported } from '../boot/isBrowserSupported'
+import { getReplayStats } from './replayStats'
+
+export function getSessionReplayLink(
+  configuration: RumConfiguration,
+  sessionManager: RumSessionManager,
+  viewContexts: ViewContexts
+): string | undefined {
+  const session = sessionManager.findTrackedSession()
+  const parameters: string[] = []
+  const sessionId = session ? session.id : 'session-id'
+
+  const view = viewContexts.findView()
+
+  const errorType = getErrorType(session, view)
+  if (errorType) {
+    parameters.push(`error-type=${errorType}`)
+  }
+  if (view) {
+    parameters.push(`seed=${view.id}`)
+    parameters.push(`from=${view.startClocks.timeStamp}`)
+  }
+
+  const origin = getDatadogOrigin(configuration)
+  const path = `/rum/replay/sessions/${sessionId}`
+  return `${origin}${path}?${parameters.join('&')}`
+}
+
+function getErrorType(session: RumSession | undefined, view: ViewContext | undefined) {
+  if (!isBrowserSupported()) {
+    return 'browser-not-supported'
+  }
+  if (!session) {
+    // possibilities:
+    // - rum sampled out
+    // - session expired (edge case)
+    return 'rum-not-tracked'
+  }
+  if (!session.sessionReplayAllowed) {
+    // possibilities
+    // - replay sampled out
+    return 'incorrect-session-plan'
+  }
+  if (view && !getReplayStats(view.id)) {
+    return 'replay-not-started'
+  }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about using recorder state directly rather than accessing replay stats?

Reply:
I will leave that to @BenoitZugmeyer

Reply:
It's a good idea. You could do something like this: https://gist.github.com/BenoitZugmeyer/b0c6a09367b4a7970211cb66c7eea5d3 .